core-bundle/src/Controller/ContentElement/PlayerController.php

```diff
@@ -59,9 +59,16 @@ protected function getResponse(FragmentTemplate $template, ContentModel $model,
             return new Response();
         }
 
+        $subtitlesFiles = [];
+
+        if ($model->addSubtitles && $filesystemItems->first()?->isVideo()) {
+            $subtitlesItems = FilesystemUtil::listContentsFromSerialized($this->filesStorage, $model->subtitlesSRC ?: '');
+            $subtitlesFiles = $this->getSourceFiles($subtitlesItems);
+        }
+
```

Comment:
IMHO it's cleaner to you move this bit to `buildVideoFigureData()`. This would also remove the double `isVideo()` check.